
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../04_build_consensus_peak_set/">
      
      
        <link rel="next" href="../06_differential_accessibility_analysis/">
      
      
      <link rel="icon" href="../../assets/images/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.14">
    
    
      
        <title>05 Quantify peaks - Introduction to NGS ATAC-seq data analysis</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.342714a4.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#quantify-peak-accessibility" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Introduction to NGS ATAC-seq data analysis" class="md-header__button md-logo" aria-label="Introduction to NGS ATAC-seq data analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Introduction to NGS ATAC-seq data analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              05 Quantify peaks
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/sib-swiss/atac-seq-training/tree/main" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/atac-seq-training
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Introduction to NGS ATAC-seq data analysis" class="md-nav__button md-logo" aria-label="Introduction to NGS ATAC-seq data analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    Introduction to NGS ATAC-seq data analysis
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sib-swiss/atac-seq-training/tree/main" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/atac-seq-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../precourse/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Precourse preparations
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../course_schedule/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Course schedule
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day1
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Day1
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../00_connect_to_server/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    00 Connect to server
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../01_filter_bam_files/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    01 Filter bam files
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../02_QC_post_alignment/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    02 QC post alignment
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../03_peak_calling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    03 Peak calling
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day2
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Day2
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_build_consensus_peak_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    04 Build consensus peak set
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    05 Quantify peaks
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    05 Quantify peaks
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    <span class="md-ellipsis">
      Overview
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#learning-objectives" class="md-nav__link">
    <span class="md-ellipsis">
      Learning Objectives
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#1-count-reads-on-peaks" class="md-nav__link">
    <span class="md-ellipsis">
      1. Count reads on peaks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="1. Count reads on peaks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-prepare-peak-annotation-for-quantification" class="md-nav__link">
    <span class="md-ellipsis">
      1.1 Prepare Peak Annotation for Quantification
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-quality-control-assessment" class="md-nav__link">
    <span class="md-ellipsis">
      2. Quality Control Assessment
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../06_differential_accessibility_analysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    06 Differential accessibility analysis
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../07_annotate_peaks/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    07 Annotate peaks
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    <span class="md-ellipsis">
      Overview
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#learning-objectives" class="md-nav__link">
    <span class="md-ellipsis">
      Learning Objectives
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#1-count-reads-on-peaks" class="md-nav__link">
    <span class="md-ellipsis">
      1. Count reads on peaks
    </span>
  </a>
  
    <nav class="md-nav" aria-label="1. Count reads on peaks">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-prepare-peak-annotation-for-quantification" class="md-nav__link">
    <span class="md-ellipsis">
      1.1 Prepare Peak Annotation for Quantification
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-quality-control-assessment" class="md-nav__link">
    <span class="md-ellipsis">
      2. Quality Control Assessment
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="quantify-peak-accessibility">Quantify Peak Accessibility</h1>
<h2 id="overview">Overview</h2>
<p>After building a consensus peak set, we need to quantify the number of reads overlapping each peak in each sample. This creates a count matrix for downstream differential accessibility analysis.</p>
<h3 id="learning-objectives">Learning Objectives</h3>
<ul>
<li>Convert peak coordinates to appropriate format for read counting</li>
<li>Use featureCounts to quantify reads in peaks across samples</li>
<li>Generate comprehensive quality control reports</li>
</ul>
<h2 id="1-count-reads-on-peaks">1. Count reads on peaks</h2>
<p>After defining a consensus peak set, we will count the number of reads that overlap each peak in each sample. This step can be seen as counting reads in annotated genes for RNA-seq data, where instead of genes, we are counting reads in peaks.</p>
<p>There are different tools that can be used for this purpose, such as <a href="subread.sourceforge.net/featureCounts.html">featureCounts</a> from the Subread package or <a href="https://bedtools.readthedocs.io/en/latest/content/tools/coverage.html">bedtools coverage</a>.  </p>
<p>In this example, we will use <code>featureCounts</code>, which is a widely used tool for counting reads in genomic features.  </p>
<h3 id="11-prepare-peak-annotation-for-quantification">1.1 Prepare Peak Annotation for Quantification</h3>
<p>FeatureCounts requires annotation in GTF or SAF (Simplified Annotation Format).<br />
The SAF format is a tab-delimited file with the following columns: <br />
<code>GeneID, Chr, Start, End, Strand</code></p>
<p>However, your consensus peak is in BED format with only 3 columns:  </p>
<div class="highlight"><pre><span></span><code><span class="c1"># Check file format</span>
head<span class="w"> </span>results/04_consensus_peaks/consensus_peaks.bed
</code></pre></div>
<p><strong>Task1: Compare BED to SAF format</strong></p>
<ul>
<li>Which are the 2 columns you are missing to convert your BED file into SAF format? </li>
<li>Can you think of a way to provide values to those fields?</li>
</ul>
<details class="answer">
<summary>Answer</summary>
<p><code>GeneID</code> and <code>Strand</code> fields are missing.<br />
Strand is missing since the dataset has no strand information.<br />
GeneID would correspond to Interval_ID, but this information has been lost while creating the consensus set, given that peaks have been merged.  </p>
<p><code>Strand</code>: Information is not mandatory, you can provide a <code>.</code> instead.<br />
<code>GeneID</code>: You can generate sequential Interval_ID with incremental numbers (e.g., &ldquo;Interval_1&rdquo;, &ldquo;Interval_2&rdquo;).  </p>
</details>
<p><strong>Task 2: Convert Consensus Peaks to SAF Format</strong></p>
<ul>
<li>Create a new folder named: <code>results/05_counts_reads</code></li>
<li>Use awk or an alternative way to convert the bed file to SAF format. Don&rsquo;t forget to add a header &ldquo;GeneID    Chr Start   End Strand&rdquo;. </li>
</ul>
<details class="info">
<summary>Hint</summary>
<p>Use <code>awk</code> to:<br />
- Add sequential peak IDs (e.g., &ldquo;Interval_1&rdquo;, &ldquo;Interval_2&rdquo;).<br />
- Convert BED coordinates to SAF format.<br />
- Add strand information (use &ldquo;.&rdquo; for unstranded).  </p>
</details>
<details class="success">
<summary>Solution</summary>
<p><div class="highlight"><pre><span></span><code># Create output directory
mkdir results/05_counts_reads

# Create SAF file with header
echo &quot;GeneID    Chr Start   End Strand&quot; &gt; results/04_consensus_peaks/consensus_peaks.saf

# Convert BED to SAF format
awk &#39;{OFS = &quot;\t&quot;} {print &quot;Interval_&quot;NR,$1,$2,$3,&quot;.&quot;}&#39; results/04_consensus_peaks/consensus_peaks.bed &gt;&gt; results/04_consensus_peaks/consensus_peaks.saf

# check the new file format
head results/04_consensus_peaks/consensus_peaks.saf
</code></pre></div>
<strong>What this does:</strong>.<br />
- <code>echo -e</code>: Creates header with tab separators.<br />
- <code>awk OFS="\t"</code>: Sets output field separator to tab.<br />
- <code>"Interval_"NR</code>: Creates unique ID using row number.<br />
- <code>$1,$2,$3</code>: BED coordinates (chr, start, end).<br />
- <code>"."</code>: Unstranded (appropriate for ATAC-seq peaks).  </p>
</details>
<p>Now we can use featureCounts to count the reads in each peak for each sample</p>
<p><strong>Task 3: Count reads in peaks</strong></p>
<ul>
<li>Run featureCounts to quantify how many filtered reads (in <code>*qc_bl_filt.sorted.bam</code>) overlap consensus peaks in each sample.  </li>
<li>Use paired-end counting mode</li>
<li>Count all filtered BAM files simultaneously</li>
</ul>
<div class="admonition featurecounts">
<p class="admonition-title">Featurecounts</p>
<p><code>-F SAF</code>: specify that the annotation file is in SAF format<br />
<code>-p</code>: specify that the input files are paired-end<br />
<code>-a</code>: specify the annotation file<br />
<code>-o</code>: specify the output file  </p>
<p>These are only some of all parameteres one can apply. You can also adjust the parameters of featureCounts based on your specific requirements, such as setting a minimum mapping quality or handling multi-mapping reads.</p>
</div>
<details class="success">
<summary>Solution</summary>
<div class="highlight"><pre><span></span><code>path_bams=&quot;results/01_filtered_bams&quot;
featureCounts -F SAF -T 2 -p -a results/04_consensus_peaks/consensus_peaks.saf -o results/05_counts_reads/feature_Counts.txt $path_bams/*qc_bl_filt.sorted.bam 2&gt; results/05_counts_reads/featureCounts.log
</code></pre></div>
</details>
<p>The output file will contain the counts of reads in each peak for each sample, which can be used for downstream analysis such as differential accessibility analysis.</p>
<p><strong>Task 4: Examine the results</strong></p>
<p>Have a look to the output, which file contains the counts matrix?:
<div class="highlight"><pre><span></span><code>ls results/05_counts_reads/
</code></pre></div></p>
<details class="answer">
<summary>Answer</summary>
<p>FeatureCounts generates:<br />
- <code>feature_Counts.txt</code>: Main count matrix.<br />
- <code>feature_Counts.txt.summary</code>: Counting statistics.<br />
- <code>featureCounts.log</code>: Processing log and any warnings.  </p>
<p>File <code>results/05_counts_reads/feature_Counts.txt</code> contains the counts table with:<br />
- Rows: Annotated peaks.<br />
- Columns: peaks annotation information (from SAF file) + Samples (counts in each sample)</p>
</details>
<ul>
<li>Check how many peaks contains the counts matrix, is it the same number you had in your consensus peak set?</li>
</ul>
<div class="highlight"><pre><span></span><code><span class="c1"># Count number of peaks quantified</span>
wc<span class="w"> </span>-l<span class="w"> </span>results/05_counts_reads/feature_Counts.txt
</code></pre></div>
<details class="answer">
<summary>Answer</summary>
<p>Yes, it is the same number of peaks, although not same number of rows because <code>feature_Counts.txt</code> has 2 lines header.</p>
</details>
<h2 id="2-quality-control-assessment">2.  Quality Control Assessment</h2>
<p>After filtering, peak calling and counting reads in peaks, we can run MultiQC to aggregate the QC metrics from different steps and generate a comprehensive report.  </p>
<p>This will help us to assess further the overall quality of the ATAC-seq data and identify any potential issues that may need to be addressed before proceeding with downstream analysis.  </p>
<p><strong>Task 3: Generate MultiQC Report</strong>  </p>
<ul>
<li>Run multiqc on the entire <code>results</code> directory</li>
</ul>
<details class="success">
<summary>Solution</summary>
<p><div class="highlight"><pre><span></span><code>multiqc --outdir results/multiQC_report --title &quot;ATAC-seq_Pipeline_Summary&quot; results/
</code></pre></div>
<strong>Parameter explanations:</strong>.<br />
- <code>--outdir</code>: Output directory for the report.<br />
- <code>--title</code>: Custom title for the report.<br />
- <code>results/</code>: Search this directory for QC files.  </p>
</details>
<p>Download and open the report and explore the QC metrics:</p>
<ul>
<li>How many peaks were successfully quantified?   </li>
<li>What percentage of reads were assigned to peaks in each sample?   </li>
<li>Are there differences in counting efficiency between samples?   </li>
<li>Do any samples show concerning quality metrics?   </li>
</ul>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.13a4f30d.min.js"></script>
      
    
  </body>
</html>