
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../06_differential_accessibility_analysis/">
      
      
      
      <link rel="icon" href="../../assets/images/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.14">
    
    
      
        <title>07 Annotate peaks - Introduction to NGS ATAC-seq data analysis</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.342714a4.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#peak-annotation-and-functional-analysis" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Introduction to NGS ATAC-seq data analysis" class="md-header__button md-logo" aria-label="Introduction to NGS ATAC-seq data analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Introduction to NGS ATAC-seq data analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              07 Annotate peaks
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/sib-swiss/atac-seq-training/tree/main" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/atac-seq-training
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Introduction to NGS ATAC-seq data analysis" class="md-nav__button md-logo" aria-label="Introduction to NGS ATAC-seq data analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    Introduction to NGS ATAC-seq data analysis
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sib-swiss/atac-seq-training/tree/main" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/atac-seq-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../precourse/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Precourse preparations
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../course_schedule/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Course schedule
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day1
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Day1
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../00_connect_to_server/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    00 Connect to server
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../01_filter_bam_files/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    01 Filter bam files
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../02_QC_post_alignment/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    02 QC post alignment
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../03_peak_calling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    03 Peak calling
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day2
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Day2
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_build_consensus_peak_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    04 Build consensus peak set
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../05_quantify_peaks/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    05 Quantify peaks
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../06_differential_accessibility_analysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    06 Differential accessibility analysis
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    07 Annotate peaks
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    07 Annotate peaks
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" class="md-nav__link">
    <span class="md-ellipsis">
      Learning Objectives
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#1-load-da-results" class="md-nav__link">
    <span class="md-ellipsis">
      1. Load DA results
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-visualize-peak-distribution" class="md-nav__link">
    <span class="md-ellipsis">
      2. Visualize Peak Distribution
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-filter-significant-peaks" class="md-nav__link">
    <span class="md-ellipsis">
      3. Filter Significant Peaks
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-annotate-genomic-overlap-with-chipseeker" class="md-nav__link">
    <span class="md-ellipsis">
      4. Annotate Genomic Overlap with ChIPseeker
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-functional-enrichment-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      5. Functional Enrichment Analysis
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-additional-visualizations" class="md-nav__link">
    <span class="md-ellipsis">
      7. Additional Visualizations
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" class="md-nav__link">
    <span class="md-ellipsis">
      Learning Objectives
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#1-load-da-results" class="md-nav__link">
    <span class="md-ellipsis">
      1. Load DA results
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-visualize-peak-distribution" class="md-nav__link">
    <span class="md-ellipsis">
      2. Visualize Peak Distribution
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-filter-significant-peaks" class="md-nav__link">
    <span class="md-ellipsis">
      3. Filter Significant Peaks
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-annotate-genomic-overlap-with-chipseeker" class="md-nav__link">
    <span class="md-ellipsis">
      4. Annotate Genomic Overlap with ChIPseeker
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-functional-enrichment-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      5. Functional Enrichment Analysis
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-additional-visualizations" class="md-nav__link">
    <span class="md-ellipsis">
      7. Additional Visualizations
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="peak-annotation-and-functional-analysis">Peak Annotation and Functional Analysis</h1>
<p>After performing differential accessibility analysis with DESeq2, you will annotate peaks based on their genomic location using ChIPseeker and perform functional enrichment analysis.</p>
<h3 id="learning-objectives">Learning Objectives</h3>
<ul>
<li>Load differential accessibility results and prepare for annotation</li>
<li>Understand genomic feature annotation using ChIPseeker</li>
<li>Visualize peak distribution and genomic context</li>
<li>Perform functional enrichment analysis on peak-associated genes </li>
<li>Interpret biological significance of accessibility changes </li>
</ul>
<p>Load necessary packages for the analysis </p>
<div class="highlight"><pre><span></span><code><span class="c1"># Load libraries</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ChIPseeker</span><span class="p">)</span>
<span class="nf">require</span><span class="p">(</span><span class="s">&quot;TxDb.Mmusculus.UCSC.mm10.knownGene&quot;</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">clusterProfiler</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="s">&quot;org.Mm.eg.db&quot;</span><span class="p">)</span>


<span class="c1"># Set up annotation databases</span>
<span class="n">TxDb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">TxDb.Mmusculus.UCSC.mm10.knownGene</span>
<span class="n">AnnoDb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&#39;org.Mm.eg.db&#39;</span>
</code></pre></div>
<h2 id="1-load-da-results">1. Load DA results</h2>
<p>Load the DA results you have generated in the previous section, and name it <code>DA_results</code> again:</p>
<details class="success">
<summary>Solution</summary>
<div class="highlight"><pre><span></span><code><span class="n">gr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readRDS</span><span class="p">(</span><span class="s">&quot;results/06_DA_analysis/DA_results.RDS&quot;</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">gr</span><span class="p">)</span>
</code></pre></div>
</details>
<h2 id="2-visualize-peak-distribution">2. Visualize Peak Distribution</h2>
<p>Let&rsquo;s visualize how peaks are distributed across the genome. In this case, they are all located at chr6, but when analysing the entire genome you may want to see if they cluster in specific chromosomes or locations. 
<code>covplot</code> function from <a href="https://academic.oup.com/bioinformatics/article/31/14/2382/255379">ChipSeeker</a> package allows you to do that: </p>
<p><strong>Task1: Plot peaks across the genome</strong></p>
<ul>
<li>Serach for <code>covplot</code> function in <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/ChIPseeker/inst/doc/ChIPseeker.html">ChipSeeker documentation</a>  and plot the genomic location of peaks, as well as their logFC. </li>
</ul>
<details class="success">
<summary>Solution</summary>
<div class="highlight"><pre><span></span><code># Create coverage plot showing peak positions and their fold changes
covplot(gr, weightCol = &quot;log2FoldChange&quot;)
</code></pre></div>
</details>
<p>This plot shows the distribution of peaks across chromosomes, with y-axis representing the log2 fold change values.</p>
<h2 id="3-filter-significant-peaks">3. Filter Significant Peaks</h2>
<p>Split peaks into upregulated and downregulated categories:</p>
<ul>
<li>Sort GRanges by genomic coordinates </li>
<li>Filter for significant peaks (padj &lt; 0.01)</li>
<li>Apply fold change thresholds (|log2FC| &gt; 2)</li>
</ul>
<div class="highlight"><pre><span></span><code><span class="c1"># Sort the GRanges object by genomic coordinates</span>
<span class="n">gr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="n">gr</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">seqnames</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">start</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">end</span><span class="p">)</span>

<span class="c1"># Split peaks into upregulated and downregulated based on significance and fold change</span>
<span class="n">gr_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span>
<span class="w">  </span><span class="n">up</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gr</span><span class="p">[</span><span class="n">gr</span><span class="o">$</span><span class="n">padj</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.01</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">gr</span><span class="o">$</span><span class="n">log2FoldChange</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">2</span><span class="p">,],</span><span class="w"> </span>
<span class="w">  </span><span class="n">down</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gr</span><span class="p">[</span><span class="n">gr</span><span class="o">$</span><span class="n">padj</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.01</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">gr</span><span class="o">$</span><span class="n">log2FoldChange</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">-2</span><span class="p">,]</span>
<span class="p">)</span>

<span class="c1"># Check the structure of gr_lists (list with 2 GRanges object)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">gr_list</span><span class="p">)</span>
</code></pre></div>
<p><strong>Task 2: Explore gr objects</strong> </p>
<ul>
<li>Check how many rows contain each object of the list, are the numbers balanced?</li>
</ul>
<details class="success">
<summary>Solution</summary>
<div class="highlight"><pre><span></span><code># Count peaks in each category
cat(&quot;Number of upregulated peaks:&quot;, length(gr_list$up), &quot;\n&quot;)
cat(&quot;Number of downregulated peaks:&quot;, length(gr_list$down), &quot;\n&quot;)
</code></pre></div>
</details>
<ul>
<li>Which tissue shows more significantly accessible regions?</li>
<li>Is this in agreement with what you have seen in the Volcano plot?</li>
</ul>
<h2 id="4-annotate-genomic-overlap-with-chipseeker">4. Annotate Genomic Overlap with ChIPseeker</h2>
<p>Next, you will annotate peaks to determine their genomic context (promoters, exons, introns, etc.):</p>
<p><strong>Annotation Parameters:</strong>.<br />
- TSS region: ±1000 bp around transcription start sites.<br />
- Include detailed genomic annotations.<br />
- Separate analysis for up and down regulated peaks. </p>
<div class="highlight"><pre><span></span><code><span class="c1"># Annotate all peaks</span>
<span class="n">peakAnno</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">annotatePeak</span><span class="p">(</span><span class="n">gr</span><span class="p">,</span><span class="w"> </span>
<span class="w">                        </span><span class="n">tssRegion</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">-1000</span><span class="p">,</span><span class="w"> </span><span class="m">1000</span><span class="p">),</span><span class="w"> </span>
<span class="w">                        </span><span class="n">TxDb</span><span class="o">=</span><span class="n">TxDb</span><span class="p">,</span><span class="w"> </span>
<span class="w">                        </span><span class="n">annoDb</span><span class="o">=</span><span class="n">AnnoDb</span><span class="p">,</span><span class="w"> </span>
<span class="w">                        </span><span class="n">overlap</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;TSS&quot;</span><span class="p">)</span>


<span class="c1"># Annotate upregulated peaks</span>
<span class="n">peakAnno_up</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">annotatePeak</span><span class="p">(</span><span class="n">gr_list</span><span class="o">$</span><span class="n">up</span><span class="p">,</span><span class="w"> </span>
<span class="w">                           </span><span class="n">tssRegion</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">-1000</span><span class="p">,</span><span class="w"> </span><span class="m">1000</span><span class="p">),</span><span class="w"> </span>
<span class="w">                           </span><span class="n">TxDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">TxDb</span><span class="p">,</span><span class="w"> </span>
<span class="w">                           </span><span class="n">annoDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">AnnoDb</span><span class="p">,</span><span class="w"> </span>
<span class="w">                           </span><span class="n">overlap</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;TSS&quot;</span><span class="p">)</span>

<span class="c1"># Annotate downregulated peaks</span>
<span class="n">peakAnno_down</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">annotatePeak</span><span class="p">(</span><span class="n">gr_list</span><span class="o">$</span><span class="n">down</span><span class="p">,</span><span class="w"> </span>
<span class="w">                             </span><span class="n">tssRegion</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">-1000</span><span class="p">,</span><span class="w"> </span><span class="m">1000</span><span class="p">),</span><span class="w"> </span>
<span class="w">                             </span><span class="n">TxDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">TxDb</span><span class="p">,</span><span class="w"> </span>
<span class="w">                             </span><span class="n">annoDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">AnnoDb</span><span class="p">,</span><span class="w"> </span>
<span class="w">                             </span><span class="n">overlap</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;TSS&quot;</span><span class="p">)</span>


<span class="c1"># Save the objects</span>
<span class="nf">saveRDS</span><span class="p">(</span><span class="n">peakAnno</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;results/06_DA_analysis/Annotated_peaks.rds&quot;</span><span class="p">)</span>
<span class="nf">saveRDS</span><span class="p">(</span><span class="n">peakAnno_up</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;results/06_DA_analysis/Annotated_peaks_up.rds&quot;</span><span class="p">)</span>
<span class="nf">saveRDS</span><span class="p">(</span><span class="n">peakAnno_down</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;results/06_DA_analysis/Annotated_peaks_down.rds&quot;</span><span class="p">)</span>
</code></pre></div>
<p><strong>Task 3: Examine Annotation Results</strong></p>
<p>Explore the new objects created after annotation.   </p>
<ul>
<li>What is the frequency of promoters overlap overall?</li>
<li>Do upregulated and downregulated peaks have different frequencies?  </li>
</ul>
<p><code>peakAnno</code>, you can find different layers of information if you do <code>peakAnno@&lt;tab&gt;</code>:</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Look at all peaks annotation summary</span>
<span class="n">peakAnno</span>
<span class="c1"># Look at the upregulated peaks annotation summary</span>
<span class="n">peakAnno_up</span>
<span class="c1"># Look at the downregulated peaks annotation summary</span>
<span class="n">peakAnno_down</span>
</code></pre></div>
<p>You can examine <code>peakAnno</code> object by doing <code>peakAnno@&lt;tab&gt;</code>, you will find several layers of information.  </p>
<ul>
<li>How many downregulated peaks overlap exons?  </li>
</ul>
<div class="highlight"><pre><span></span><code><span class="c1"># Count how many downregulated peaks overlap exonic regions</span>
<span class="nf">sum</span><span class="p">(</span><span class="n">peakAnno_down</span><span class="o">@</span><span class="n">detailGenomicAnnotation</span><span class="o">$</span><span class="n">Exon</span><span class="p">)</span>
</code></pre></div>
<p><strong>Task 4: Visualise Peak Annotations</strong></p>
<p>Use <code>plotAnnoPie</code> function from ChipSeeker package to visualize the genomic overlap distribution of upregulated peaks and downregulated peaks, separately:</p>
<details class="success">
<summary>Solution</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># Plot pie charts for peak categories</span>
<span class="nf">plotAnnoPie</span><span class="p">(</span><span class="n">peakAnno_up</span><span class="p">,</span><span class="w"> </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\n\nDA up&quot;</span><span class="p">)</span><span class="w"> </span>
<span class="nf">plotAnnoPie</span><span class="p">(</span><span class="n">peakAnno_down</span><span class="p">,</span><span class="w"> </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\n\nDA down&quot;</span><span class="p">)</span>
</code></pre></div>
</details>
<p>These plots show the percentage of peaks falling into different genomic categories (promoters, exons, introns, intergenic regions, etc.).</p>
<h2 id="5-functional-enrichment-analysis">5. Functional Enrichment Analysis</h2>
<p>Next, we will focus in peaks overlaping promoter regions (TSS) of genes, and we will perform functional enrichment analysis on those genes to understand which biological processes or metabolic pathways may be afected by the changes in chromatin accessibility.</p>
<p><strong>Task 6: Prepare gene list</strong></p>
<p>Start testing DA peaks that are upregulated (higher accessibility in Kidney).  </p>
<ul>
<li>Take DA upregulated peaks (peakAnno_up), select those overlapping TSS (Promoter), and get the list of gene SYMBOLs of the overlapping genes.  </li>
<li>Get a list of universe gene SYMBOLs to compare the DA upregulated genes to.  </li>
</ul>
<details class="success">
<summary>Solution</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># Extract gene symbols for peaks in promoter regions</span>
<span class="n">genes_tss_up</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">peakAnno_up</span><span class="o">@</span><span class="n">anno</span><span class="p">[</span><span class="n">peakAnno_up</span><span class="o">@</span><span class="n">detailGenomicAnnotation</span><span class="o">$</span><span class="n">Promoter</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span><span class="p">]</span>

<span class="c1"># Create universe of genes with all promoter-overlapping peaks for background</span>
<span class="n">universe</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">peakAnno</span><span class="o">@</span><span class="n">anno</span><span class="p">[</span><span class="n">peakAnno</span><span class="o">@</span><span class="n">detailGenomicAnnotation</span><span class="o">$</span><span class="n">Promoter</span><span class="p">,]</span><span class="o">$</span><span class="n">SYMBOL</span>
</code></pre></div>
</details>
<p><strong>Task 7: Gene Ontology (GO) Enrichment Analysis</strong></p>
<p>Use <code>enrichGO</code> function from <a href="https://bioconductor.org/packages/devel/bioc/manuals/clusterProfiler/man/clusterProfiler.pdf"><code>ClusterProfiler</code></a> to perform GO Enrichment Analysis of <code>genes_tss_up</code> set.   </p>
<ul>
<li>Provide unique universe. </li>
<li>Choose Biological Process (BP) ontology.  </li>
<li>Apply multiple testing correction, and pvalueCutoff 0.05</li>
<li>Pass AnnoDb variable created at the beginning of this tutorial section for OrgDb.  </li>
<li>Specify keyType (SYMBOL).  </li>
</ul>
<details class="success">
<summary>Solution</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># GO enrichment for upregulated genes</span>
<span class="n">ego_up</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span><span class="n">gene</span><span class="w">          </span><span class="o">=</span><span class="w"> </span><span class="n">genes_tss_up</span><span class="o">$</span><span class="n">SYMBOL</span><span class="p">,</span>
<span class="w">                </span><span class="n">universe</span><span class="w">      </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">universe</span><span class="p">),</span>
<span class="w">                </span><span class="n">OrgDb</span><span class="w">         </span><span class="o">=</span><span class="w"> </span><span class="n">AnnoDb</span><span class="p">,</span>
<span class="w">                </span><span class="n">keyType</span><span class="w">       </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span>
<span class="w">                </span><span class="n">ont</span><span class="w">           </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BP&quot;</span><span class="p">,</span><span class="w">  </span>
<span class="w">                </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">,</span>
<span class="w">                </span><span class="n">pvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">                </span><span class="n">qvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">                </span><span class="n">readable</span><span class="w">      </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>

<span class="c1"># Check results</span>
<span class="n">ego_up</span>
</code></pre></div>
</details>
<p><strong>Task 8</strong>  </p>
<ul>
<li>
<p>Did you find significantly enriched GO terms?  </p>
</li>
<li>
<p>Try to run the same analysis considering all genes as a universe, why do you think there is a difference?  </p>
</li>
<li>
<p>Visualise the GO terms using a function from <a href="https://yulab-smu.top/biomedical-knowledge-mining-book/enrichplot.html">ClusterProfiler</a> such as: <code>barplot()</code></p>
</li>
</ul>
<details class="success">
<summary>Solution</summary>
<div class="highlight"><pre><span></span><code><span class="nf">barplot</span><span class="p">(</span><span class="n">ego_up</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">)</span>
</code></pre></div>
</details>
<div class="admonition tip">
<p class="admonition-title">Bonus</p>
<p>Repeat tasts 6-8 for DA downregulated peaks (higher accessiblility in Cerebrum)</p>
<details class="success">
<summary>Solution</summary>
<div class="highlight"><pre><span></span><code><span class="c1"># Create list of gene SYMBOLS associated to TSS overlapping peaks</span>
<span class="n">genes_tss_down</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">peakAnno_down</span><span class="o">@</span><span class="n">anno</span><span class="p">[</span><span class="n">peakAnno_down</span><span class="o">@</span><span class="n">detailGenomicAnnotation</span><span class="o">$</span><span class="n">Promoter</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span><span class="p">]</span>

<span class="c1"># Run enrichGO using gene SYMBOLS associated to all peaks overlapping promoters</span>
<span class="n">ego_down</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span><span class="n">gene</span><span class="w">   </span><span class="o">=</span><span class="w"> </span><span class="n">genes_tss_down</span><span class="o">$</span><span class="n">SYMBOL</span><span class="p">,</span>
<span class="w">            </span><span class="n">universe</span><span class="w">      </span><span class="o">=</span><span class="w"> </span><span class="nf">unique</span><span class="p">(</span><span class="n">universe</span><span class="p">),</span>
<span class="w">            </span><span class="n">OrgDb</span><span class="w">         </span><span class="o">=</span><span class="w"> </span><span class="n">AnnoDb</span><span class="p">,</span>
<span class="w">            </span><span class="n">keyType</span><span class="w">       </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span>
<span class="w">            </span><span class="n">ont</span><span class="w">           </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BP&quot;</span><span class="p">,</span><span class="w">  </span>
<span class="w">            </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">,</span>
<span class="w">            </span><span class="n">pvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">            </span><span class="n">qvalueCutoff</span><span class="w">  </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">            </span><span class="n">readable</span><span class="w">      </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>

<span class="c1"># Plot results</span>
<span class="nf">barplot</span><span class="p">(</span><span class="n">ego_down</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">)</span>
</code></pre></div>
</details>
</div>
<div class="admonition tip">
<p class="admonition-title">Next Steps</p>
<p>You can change the <code>ont</code> parameter to explore different ontologies:<br />
- &ldquo;MF&rdquo;: Molecular Function.<br />
- &ldquo;BP&rdquo;: Biological Process <br />
- &ldquo;CC&rdquo;: Cellular Component  </p>
</div>
<h2 id="7-additional-visualizations">7. Additional Visualizations</h2>
<p>There are several other plots that ChipSeeker allowes you to do. 
You can explore a couple of examples related to plotting the profile of peaks around certain regions</p>
<p><strong>Task 9: Peak Heatmap</strong></p>
<ul>
<li>Create a heatmap showing peak signal around TSS:  </li>
</ul>
<div class="highlight"><pre><span></span><code><span class="c1"># Generate peak heatmap around gene bodies</span>
<span class="nf">peakHeatmap</span><span class="p">(</span><span class="n">peak</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gr</span><span class="p">,</span>
<span class="w">            </span><span class="n">TxDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">TxDb</span><span class="p">,</span>
<span class="w">            </span><span class="n">upstream</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1000</span><span class="p">,</span>
<span class="w">            </span><span class="n">downstream</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1000</span><span class="p">,</span>
<span class="w">            </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;gene&quot;</span><span class="p">,</span>
<span class="w">            </span><span class="n">type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;start_site&quot;</span><span class="p">,</span>
<span class="w">            </span><span class="n">nbin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">800</span><span class="p">)</span>
</code></pre></div>
<div class="admonition tip">
<p class="admonition-title">Bonus Task</p>
<p>Now let&rsquo;s plot the profile of peak counts around annotated TSS split by conditions</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Get TSS (promoter) annotation for mouse mm10</span>
<span class="n">promoter</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getPromoters</span><span class="p">(</span><span class="n">TxDb</span><span class="o">=</span><span class="n">TxDb</span><span class="p">,</span><span class="w"> </span><span class="n">upstream</span><span class="o">=</span><span class="m">1000</span><span class="p">,</span><span class="w"> </span><span class="n">downstream</span><span class="o">=</span><span class="m">1000</span><span class="p">)</span>

<span class="c1"># Build a matrix of peaks per conditions, and combine them in a list</span>
<span class="n">tagMatrix_Cer</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getTagMatrix</span><span class="p">(</span><span class="n">gr_list</span><span class="o">$</span><span class="n">down</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">windows</span><span class="o">=</span><span class="n">promoter</span><span class="p">)</span>
<span class="n">tagMatrix_Kid</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getTagMatrix</span><span class="p">(</span><span class="n">gr_list</span><span class="o">$</span><span class="n">up</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">windows</span><span class="o">=</span><span class="n">promoter</span><span class="p">)</span>
<span class="n">tagMatrixList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">Cerebrum</span><span class="o">=</span><span class="n">agMatrix_Cer</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">Kidney</span><span class="o">=</span><span class="n">tagMatrix_Kid</span><span class="w"> </span><span class="p">)</span>

<span class="c1"># plot </span>
<span class="nf">plotAvgProf</span><span class="p">(</span><span class="n">tagMatrixList</span><span class="p">,</span><span class="w"> </span><span class="n">xlim</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">-1000</span><span class="p">,</span><span class="w"> </span><span class="m">1000</span><span class="p">),</span><span class="w"> </span><span class="n">conf</span><span class="o">=</span><span class="m">0.95</span><span class="p">,</span><span class="n">resample</span><span class="o">=</span><span class="m">500</span><span class="p">,</span><span class="w"> </span><span class="n">facet</span><span class="o">=</span><span class="s">&quot;row&quot;</span><span class="p">)</span>
</code></pre></div>
</div>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.13a4f30d.min.js"></script>
      
    
  </body>
</html>