
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../00_connect_to_server/">
      
      
        <link rel="next" href="../02_QC_post_alignment/">
      
      
      <link rel="icon" href="../../assets/images/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.14">
    
    
      
        <title>01 Filter bam files - Introduction to NGS ATAC-seq data analysis</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.342714a4.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#post-alignment-reads-filtering" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Introduction to NGS ATAC-seq data analysis" class="md-header__button md-logo" aria-label="Introduction to NGS ATAC-seq data analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Introduction to NGS ATAC-seq data analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              01 Filter bam files
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/sib-swiss/atac-seq-training/tree/main" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/atac-seq-training
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Introduction to NGS ATAC-seq data analysis" class="md-nav__button md-logo" aria-label="Introduction to NGS ATAC-seq data analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    Introduction to NGS ATAC-seq data analysis
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sib-swiss/atac-seq-training/tree/main" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/atac-seq-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../precourse/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Precourse preparations
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../course_schedule/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Course schedule
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" checked>
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day1
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Day1
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../00_connect_to_server/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    00 Connect to server
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    01 Filter bam files
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    01 Filter bam files
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    <span class="md-ellipsis">
      Overview
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#learning-objectives" class="md-nav__link">
    <span class="md-ellipsis">
      Learning Objectives
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#0-dataset" class="md-nav__link">
    <span class="md-ellipsis">
      0. Dataset
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#1-filtering-reads-from-alignments" class="md-nav__link">
    <span class="md-ellipsis">
      1. Filtering Reads from Alignments
    </span>
  </a>
  
    <nav class="md-nav" aria-label="1. Filtering Reads from Alignments">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-common-ngs-filtering-steps" class="md-nav__link">
    <span class="md-ellipsis">
      1.1 Common NGS Filtering Steps
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#12-atacseq-related-filtering-steps" class="md-nav__link">
    <span class="md-ellipsis">
      1.2 ATACseq related filtering steps
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../02_QC_post_alignment/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    02 QC post alignment
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../03_peak_calling/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    03 Peak calling
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Day2
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Day2
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_build_consensus_peak_set/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    04 Build consensus peak set
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../05_quantify_peaks/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    05 Quantify peaks
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../06_differential_accessibility_analysis/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    06 Differential accessibility analysis
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../07_annotate_peaks/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    07 Annotate peaks
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    <span class="md-ellipsis">
      Overview
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Overview">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#learning-objectives" class="md-nav__link">
    <span class="md-ellipsis">
      Learning Objectives
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#0-dataset" class="md-nav__link">
    <span class="md-ellipsis">
      0. Dataset
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#1-filtering-reads-from-alignments" class="md-nav__link">
    <span class="md-ellipsis">
      1. Filtering Reads from Alignments
    </span>
  </a>
  
    <nav class="md-nav" aria-label="1. Filtering Reads from Alignments">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-common-ngs-filtering-steps" class="md-nav__link">
    <span class="md-ellipsis">
      1.1 Common NGS Filtering Steps
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#12-atacseq-related-filtering-steps" class="md-nav__link">
    <span class="md-ellipsis">
      1.2 ATACseq related filtering steps
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="post-alignment-reads-filtering">Post-Alignment Reads Filtering</h1>
<h2 id="overview">Overview</h2>
<p>In this section, we will filter BAM files to remove low-quality reads and prepare them for downstream analysis.</p>
<h3 id="learning-objectives">Learning Objectives</h3>
<ul>
<li>Apply quality filters to BAM files using samtools</li>
<li>Remove reads from blacklisted genomic regions</li>
<li>Understand ATAC-seq specific filtering considerations</li>
</ul>
<h2 id="0-dataset">0. Dataset</h2>
<p>We are going to work with a subset of the publicly available ATAC-seq dataset from <a href="https://www.nature.com/articles/s41597-019-0071-0">Liu et al. 2019</a>. We will process and compare data from 2 different adult mouse tissues: </p>
<ul>
<li><strong>Kidney</strong>: Rep1, Rep2     </li>
<li><strong>Cerebrum</strong>: Rep1, Rep2  </li>
</ul>
<p>In order to save time, raw reads have already been trimmed using <a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">Trim Galore</a> and mapped to the reference genome (mm10) using <a href="https://bowtie-bio.sourceforge.net/bowtie2/index.shtml">bowtie2</a> in end-to-end mode. Therefore, we will start the analysis directly from the alignment output (<strong>.bam</strong> files).  </p>
<p>To avoid large waiting times during high-demanding computational steps, .bam files have been subset to keep only reads aligning to chromosome 6.  </p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>You can find the .bam files in <code>/data/Liu_alignments_chr6/</code> folder.<br />
Raw data can be found in <a href="https://www.ncbi.nlm.nih.gov/sra/SRP167062">NCBI Sequence Read Archive</a>.</p>
</div>
<h2 id="1-filtering-reads-from-alignments">1. Filtering Reads from Alignments</h2>
<h3 id="11-common-ngs-filtering-steps">1.1 Common NGS Filtering Steps</h3>
<p>To ensure high-quality data, it&rsquo;s important to filter aligned reads based on standard NGS criteria, such as:</p>
<ul>
<li>Removing duplicate reads (often PCR artifacts).  </li>
<li>Applying minimum mapping quality thresholds.  </li>
</ul>
<p>These filters help reduce noise and improve the accuracy of downstream analyses.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Tools like Picard can identify and mark duplicate reads. In this dataset, duplicates have already been marked in the provided .bam files. To learn more about Picard have a look <a href="https://broadinstitute.github.io/picard/command-line-overview.html#MarkDuplicates">here</a>.   </p>
</div>
<p>A useful tool for filtering alignments is <a href="https://www.htslib.org/doc/samtools-view.html"><code>samtools view</code></a>. It allows printing all the alignments from a .bam file in SAM format, with the option to previously filter them based on specific flags (e.g. mapped/unmapped, primary/secondary, duplicates).  </p>
<p>In the next exercises, you will use samtools view to visualise the alignment output and filter .bam files according to various criteria.</p>
<p><strong>Task 1: Visualise alignment output</strong> </p>
<ul>
<li>Visualise one of the .bam files with samtools running the following command:</li>
</ul>
<p><div class="highlight"><pre><span></span><code>samtools view /data/Liu_alignments_chr6/Kidney_rep1.bam | head
</code></pre></div>
Can you recognise what the different columns represent?<br />
Why the 3rd column contains only number 6?<br />
What is the 2nd column? </p>
<details class="done">
<summary>Answer</summary>
<p>The columns correspond to SAM format. You can find an explanation of each column <a href="https://www.htslib.org/doc/sam.html">here</a>.<br />
The 3rd column represents the chromosome name. In this tutorial, .bam files have been subset to contain only chr6, thus it makes sense they are all the same.<br />
The 2nd column contains the FLAG, you can use this value to filter out or in certain reads.  </p>
</details>
<p><strong>Task 2: Quality filtering with samtools</strong> </p>
<p>Check in this <a href="https://broadinstitute.github.io/picard/explain-flags.html">website</a>, which FLAG number  would you need in order to filter out from the alignments:  </p>
<ul>
<li>Unmapped reads</li>
<li>Reads who&rsquo;s mate is unmapped</li>
<li>Read duplicates</li>
<li>Not primary alignment reads</li>
</ul>
<details class="done">
<summary>Answer</summary>
<p>You need flag: 1292</p>
</details>
<p>Now, use samtools view <a href="https://www.htslib.org/doc/samtools-view.html">manual</a> to filter the previous .bam file based on all these criteria:  </p>
<ul>
<li>Filter out reads with flag 1292.  </li>
<li>Keep only paired reads.  </li>
<li>Filter out reads with mapping quality &lt; 10.  </li>
</ul>
<details>
<summary>Hint</summary>
You can use:   
    -f and -F flags to filter in or out reads (respectively) based on a combination of SAM Flags; use -q: for MapQ threshold
</details>

<details class="success">
<summary>Solution</summary>
<div class="highlight"><pre><span></span><code>samtools view  -q 10 -f 1 -F 1292 /data/Liu_alignments_chr6/Kidney_rep2.bam | head
</code></pre></div>
</details>
<p>Finally, apply these filters to all .bam files and save them in .bam format (check parameter -b for that) in <code>results/01_filtered_bams/&lt;sample_name&gt;.qc_filt.bam</code></p>
<details class="success">
<summary>Solution</summary>
<div class="highlight"><pre><span></span><code># create new directory for filtered bams

mkdir -p results/01_filtered_bams

# filter files using a for loop

for bam in /data/Liu_alignments_chr6/*.bam; do
    echo &quot;Processing file: $bam&quot;
    sample_name=$(basename &quot;$bam&quot; .bam) # extract sample name without path and extension
    samtools view -h -b -q 10 -f 1 -F 1292 -o results/01_filtered_bams/$sample_name.qc_filt.bam $bam
done
</code></pre></div>
<ul>
<li><strong>-h</strong>: keep header  </li>
<li><strong>-b</strong>: output in bam format  </li>
<li><strong>-q 10</strong>: minimum mapping quality 10  </li>
<li><strong>-f 1</strong>: keep paired  </li>
<li><strong>-F 1292</strong>: exclude reads with any of the following flags: read unmapped, mate unmapped, not primary alignment, read is duplicate</li>
</ul>
</details>
<p>After filtering we will sort and index the bam files for the next step</p>
<p><strong>Task 3: Sort and index BAM files</strong> </p>
<p>Next, sort and index the bam files for downstream analysis.</p>
<div class="highlight"><pre><span></span><code>for bam in results/01_filtered_bams/*qc_filt.bam; do
    echo &quot;Processing file: $bam&quot;
    samtools sort -o ${bam%.bam}.sorted.bam $bam
    samtools index ${bam%.bam}.sorted.bam
done
</code></pre></div>
<p>To avoid confusion and large size files, keep only the sorted and indexed bams:</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p><strong>Important before deleting:</strong><br />
Run this command <em>only</em> if you have named your files exactly as specified in this tutorial and have completed all sorting and indexing steps from the previous command.</p>
</div>
<div class="highlight"><pre><span></span><code>rm results/01_filtered_bams/*qc_filt.bam
</code></pre></div>
<h3 id="12-atacseq-related-filtering-steps">1.2 ATACseq related filtering steps</h3>
<p>Mitochondria DNA is nucleosome free, therefore it is more accessible for Tn5 and several reads may have originated from mitochondrial DNA. After having assessed mitochondrial % on the QC, we can discard reads coming from chmMT to avoid biases in downstream analysis.</p>
<p>Since we are working only with chr6 we don&rsquo;t need to do this step, but here is the command you could use for that:
<div class="highlight"><pre><span></span><code>samtools view -h input.bam | awk  &#39;($3 != &quot;MT&quot;)&#39; | samtools view -hb - &gt; output.bam
</code></pre></div></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The mitochondrial chromosome name may differ depending on the reference genome (e.g., &ldquo;MT&rdquo;, &ldquo;chrM&rdquo;, &ldquo;chrMT&rdquo;).</p>
</div>
<p>Next, we will remove reads overlapping problematic regions of the genome. ENCODE consortium has created comprehensive lists of such regions (anomalous, unstructured or high signal in NGS experiments) for different genome species (including mouse mm10). These lists are called ENCODE Blacklists, and you can find them <a href="https://github.com/Boyle-Lab/Blacklist/">here</a>. </p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The regions for mm10 have been dowloaded as a .bed file, you can find it here: <code>/data/references/mm10-blacklist.v2.nochr.bed</code></p>
</div>
<p><strong>Task 3: Remove blacklist regions</strong></p>
<ul>
<li>Using <a href="https://bedtools.readthedocs.io/en/latest/content/tools/intersect.html"><code>bedtools intersect</code></a> filter out reads overlapping regions in the Blacklist .bed file</li>
<li>Use previously filtered .bam files as input:<br />
<code>results/01_filtered_bams/*qc_filt.sorted.bam</code>, don&rsquo;t forget to specify your input is in BAM format</li>
<li>Use the <code>/data/references/mm10-blacklist.v2.nochr.bed</code> as regions to filter out reads from (it is already sorted)</li>
<li>Save the results in <code>results/01_filtered_bams/</code> in BAM format with the following output name: <code>&lt;sample_name&gt;.qc_bl_filt.bam</code> </li>
</ul>
<div class="admonition bedtools intersect useful information">
<p class="admonition-title">Bedtools</p>
<p><code>Bedtools intersect</code> allows one to screen for overlaps between two sets of genomic features/regions, and then decide on which kind of information do you want to report.<br />
Here, we will intersect:<br />
a) Aligned reads to the genome (filtered and sorted .bam files)<br />
b) Problematic regions listed in Blacklist .bed file<br />
We do not want to keep the reads that overlap Blacklist regions.<br />
You can find documentation on which parameters to use <a href="https://bedtools.readthedocs.io/en/latest/content/tools/intersect.html">here</a></p>
</div>
<details class="success">
<summary>Solution</summary>
<div class="highlight"><pre><span></span><code>blacklist=&quot;/data/references/mm10-blacklist.v2.nochr.bed&quot;

for bam in results/01_filtered_bams/*qc_filt.sorted.bam; do
    echo &quot;Processing file: $bam&quot;
    sample_name=$(basename &quot;$bam&quot; .qc_filt.sorted.bam) 
    bedtools intersect -v -abam $bam -b $blacklist &gt; results/01_filtered_bams/$sample_name.qc_bl_filt.bam
done
</code></pre></div>
<p>Parameter explanation:<br />
<strong>-v</strong>: only report those entries in A that have no overlap with B<br />
<strong>-abam</strong>: input is in bam format </p>
</details>
<p><strong>Task 4: Sort and index the previous files</strong>  </p>
<p>Sort and index the bam files for downstream analysis.  </p>
<div class="highlight"><pre><span></span><code>for bam in results/01_filtered_bams/*qc_bl_filt.bam; do
    echo &quot;Processing file: $bam&quot;
    sample_name=$(basename &quot;$bam&quot; .qc_bl_filt.bam) 
    samtools sort -o results/01_filtered_bams/$sample_name.qc_bl_filt.sorted.bam results/01_filtered_bams/$sample_name.qc_bl_filt.bam
    samtools index results/01_filtered_bams/$sample_name.qc_bl_filt.sorted.bam
done
</code></pre></div>
<p>After sorting the .bam files, we don&rsquo;t need the unsorted .bam. To free some space we will remove the unsorted bam files (intermediate files)
.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p><strong>Important before deleting:</strong><br />
Run this command <em>only</em> if you have named your files exactly as specified in this tutorial and have completed all sorting and indexing steps from the previous command.</p>
</div>
<div class="highlight"><pre><span></span><code>rm results/01_filtered_bams/*qc_filt.bam
rm results/01_filtered_bams/*qc_bl_filt.bam
</code></pre></div>
<div class="admonition tip">
<p class="admonition-title">Bonus Task</p>
<p><strong>Taks 5: Convert .bam file into .bigwig format</strong></p>
<p>In order to visualise ATAC-seq coverage, you can convert .bam files into .bigwig format, used to represent coverage tracks in genome browsers like IGV.<br />
A tool that allows you to do this conversion is <a href="https://deeptools.readthedocs.io/en/develop/content/tools/bamCoverage.html"><code>deeptools bamCoverage</code></a></p>
<p>Next, convert the sorted and filterd .bams into .bigwigs</p>
<div class="highlight"><pre><span></span><code>mkdir results/01_filtered_bams/filt_bigwigs
for bam in results/01_filtered_bams/*qc_bl_filt.sorted.bam; do
    echo &quot;Processing file: $bam&quot;
    sample_name=$(basename &quot;$bam&quot; .qc_bl_filt.sorted.bam) 
    bamCoverage -b $bam -o results/01_filtered_bams/filt_bigwigs/$sample_name.bw --region 6:1500000:20000000 --normalizeUsing CPM
done
</code></pre></div>
<p>Download the .bigwigs and load them into IGV. 
Remember to use mm10 genome.  </p>
<p>Look into Asns gene, which tissue has higher DNA accessibility? </p>
</div>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.13a4f30d.min.js"></script>
      
    
  </body>
</html>